### PR TITLE
Fix doctrine/persistence compatibility

### PR DESCRIPTION
**Pull request by enumag**
Closes #32

This is the same as #32, just rebased on top of v1.1 to fix conflicts. ~~The tests are passing with both doctrine/persistence <1.3 and >=1.3. Is there anything else left to do here before merging? Or should I bump it to only support the new naming?~~

Updated to require doctrine/persistence 2.0.

Also added PHP 8.0 to CI.

### CODE REVIEW

**Comment by orklah** [comment]
I don't have a good enough understanding of the test coverage to be sure there won't be impacts on projects using older versions of doctrine

However, given the low activity on this plugin, especially on older stubs, I don't think it's worth trying to keep everything cross compatible with every version in order to ensure older projects will have newer features.

IMHO, we'd be better off if we released this as a 2.0.0 with doctrine/persistence bumped to 2.0.0. If we keep 1.X branch alive, we'll still be able to provide occasional modification.

@weirdan Are you OK with that?

**Comment by weirdan** [comment]
@orklah sounds good to me

**Comment by orklah** [comment]
@enumag can you bump doctrine/persistence please?

**Comment by enumag** [comment]
Sure, I'll try to update it tomorrow.

**Comment by enumag** [comment]
@orklah @weirdan Done. I also added PHP 8.0 to CI.

File: src/Plugin.php [hunk 1]
```diff
@@ -54,13 +54,6 @@ private function getBundleStubs(): array
             return [];
         }
 
-        if (
```
Comment (orklah) [inline src/Plugin.php:50]: shouldn't it be included all the time now that we can't have less than 1.3.0?

Comment (enumag) [reply src/Plugin.php:50]: Yeah, that's why I moved the contents of `persistence-1.3+/ServiceEntityRepository.phpstub` into `ServiceEntityRepository.phpstub`.

Comment (orklah) [reply src/Plugin.php:50]: Right, I didn't notice that. Can you remove the file and the directory in your PR then?

Comment (enumag) [reply src/Plugin.php:50]: I already did that. 😅

Comment (orklah) [reply src/Plugin.php:50]: Not used to see deletions in github diff sorry 😅

**Comment by enumag** [comment]
Anything else guys? :-)

**Comment by orklah** [comment]
I'd like to see #97 resolved before we change default branch but the PR seems good in itself :)

**Review by VincentLanglet — APPROVED** [review]

**Comment by enumag** [comment]
@orklah Is someone working on #97 though? It seems stalled to me. If that's the case it's not worth waiting for in my opinion.

**Comment by orklah** [comment]
The issue I have is that we decided here to do a `2.0.0` version with only newer versions of libraries allowed. This means some users will be stuck on `1.0.0` if they can't update their dependencies. #97 is a bug that impacts `1.0.0` so I would like to see it fixed before we can start `2.0.0`.

Switching to `2.0.0` does not means we won't be able to change things in `1.0.0` but I very much hope this will be very rare. So creating the `2.0.0` while there is still a big issue on `1.0.0` does not look like a good idea.

**Comment by enumag** [comment]
I realize all that. But if nobody is working on the issue I don't think we should stall the 2.0 release for weeks or months. We can always backport the fix to the 1.0 branch when it's done.

**Comment by VincentLanglet** [comment]
I tried to make the bugfix https://github.com/psalm/psalm-plugin-doctrine/pull/99

**Comment by orklah** [comment]
Can you rebase onto 2.X please?

**Comment by enumag** [comment]
@orklah done

**Comment by orklah** [comment]
Merged, thanks. Can some of you try using branch 2.X on your projects and confirm everything is ok before I make the release?

**Comment by enumag** [comment]
Works fine for me.

**Pull request merged**